src/main/scala/mesosphere/marathon/state/PortDefinition.scala

```diff
@@ -14,7 +14,7 @@ case class PortDefinition(
 
 object PortDefinition {
   implicit val portDefinitionValidator = validator[PortDefinition] { portDefinition =>
-    portDefinition.protocol is oneOf(DiscoveryInfo.Port.AllowedProtocols)
+    portDefinition.protocol.split(',').toIterable is every(oneOf(DiscoveryInfo.Port.AllowedProtocols))
```

Comment:
In portMappings we accept `tcp,udp` as string, see https://github.com/mesosphere/marathon/blob/master/src/main/scala/mesosphere/marathon/state/Container.scala#L103